src/linear_algebra/matrix/bilinear_form.lean

```diff
@@ -0,0 +1,547 @@
+/-
+Copyright (c) 2018 Andreas Swerdlow. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andreas Swerdlow, Kexing Ying
+-/
+
+import linear_algebra.matrix.basis
+import linear_algebra.matrix.nondegenerate
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.matrix.to_linear_equiv
+import linear_algebra.bilinear_form
+
+/-!
+# Bilinear form
+
+This file defines the conversion between bilinear forms and matrices.
+
+## Main definitions
+
+ * `matrix.to_bilin` given a basis define a bilinear form
+ * `matrix.to_bilin'` define the bilinear form on `n → R`
+ * `bilin_form.to_matrix`: calculate the matrix coefficients of a bilinear form
+ * `bilin_form.to_matrix'`: calculate the matrix coefficients of a bilinear form on `n → R`
+
+## Tags
+
+bilinear_form, matrix, basis
+
+-/
+
+variables {R : Type*} {M : Type*} [semiring R] [add_comm_monoid M] [module R M]
+variables {R₁ : Type*} {M₁ : Type*} [ring R₁] [add_comm_group M₁] [module R₁ M₁]
+variables {R₂ : Type*} {M₂ : Type*} [comm_semiring R₂] [add_comm_monoid M₂] [module R₂ M₂]
+variables {R₃ : Type*} {M₃ : Type*} [comm_ring R₃] [add_comm_group M₃] [module R₃ M₃]
+variables {V : Type*} {K : Type*} [field K] [add_comm_group V] [module K V]
+variables {B : bilin_form R M} {B₁ : bilin_form R₁ M₁} {B₂ : bilin_form R₂ M₂}
```

Comment:
Can you copy across the docstring from the other file that describes the conventions here?